### PR TITLE
Fix placement for popups with custom positioning

### PR DESCRIPTION
**Pull request by kryksyh**
Fix placement for popups with custom positioning (Capo/Tuning/etc setting)

Resolves: #23935

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

File: src/framework/uicomponents/view/popupview.cpp [hunk 1]
```diff
@@ -747,20 +747,22 @@ void PopupView::updateGeometry()
     bool canFitAbove = viewRect.height() < parentTopLeft.y();
     bool canFitBelow = viewRect.bottom() < anchorRect.bottom();
 
-    if (canFitAbove && (placement() == Above || !canFitBelow)) {
-        // place above the parent
-        qreal newY = parentTopLeft.y() - viewRect.height();
-        movePos(m_globalPos.x(), newY);
+    if (placement() == Placement::Above) {
+        movePos(m_globalPos.x(), parentTopLeft.y() - viewRect.height());
         setOpensUpward(true);
```
Comment (igorkorsukov) [inline src/framework/uicomponents/view/popupview.cpp:752]: What happens if there is no space at the top?

File: src/framework/uicomponents/view/popupview.h [hunk 2]
```diff
@@ -100,22 +100,23 @@ class PopupView : public QObject, public QQmlParserStatus, public Injectable, pu
...
         CloseOnPressOutsideParent = 0x00000001,
     };
     Q_DECLARE_FLAGS(ClosePolicies, ClosePolicy)
     Q_FLAG(ClosePolicies)
 
-    enum Placement {
+    enum class Placement {
+        Default,
         Below,
         Above
```
Comment (igorkorsukov) [inline src/framework/uicomponents/view/popupview.h:121]: maybe better to name it Placement::PreferBelow Placement::PreferAbove

File: src/framework/uicomponents/view/popupview.cpp (hunk 1 above)
Comment (kryksyh) [reply src/framework/uicomponents/view/popupview.cpp:752]: Additional checks added

File: src/framework/uicomponents/view/popupview.h (hunk 2 above)
Comment (kryksyh) [reply src/framework/uicomponents/view/popupview.h:121]: Agree, that makes more sense

**Review by igorkorsukov — APPROVED** [review]

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
https://github.com/musescore/MuseScore/issues/23935 FIXED

**Pull request merged**